### PR TITLE
Add iOS support

### PR DESCRIPTION
**Pull request by krjw-eyev**
I have added iOS support. I didn't know how you versioned things so I increased it by one. pubspec.lock is broken since I also changed something in dart_vlc_core. This probably needs changes from you since I don't know how you handle updates.

I hope everything else is fine.

### CODE REVIEW

**Comment by alexmercerind** [comment]
@jnschulze I've requested you for the review. I didn't know iOS can link libVLC that easy. Seems I should work on Android too.

I assume you're busy with your schedule & life. I just wanted to ask if we could just use `"BGRA"` for macOS #122 without having to add additional features of defining the color space etc. It would just be a easy go, I believe most of VideoOutlet etc. is already made by you in that PR.

EDIT: Seems actions are failing. Don't worry I will have a look.

**Comment by alexmercerind** [comment]
Thanks for your contribution to the project @krjw-eyev. Really glad to see it working on iOS.

**Comment by krjw-eyev** [comment]
It's a pleasure! I use it on iPadOS and it works great but I think the UI code of the example probably breaks on iPhone.

**Comment by krjw-eyev** [comment]
Fixed UI for iPhone

**Comment by jnschulze** [comment]
> @jnschulze I've requested you for the review. I didn't know iOS can link libVLC that easy. Seems I should work on Android too.
> 
> I assume you're busy with your schedule & life. I just wanted to ask if we could just use `"BGRA"` for macOS #122 without having to add additional features of defining the color space etc. It would just be a easy go, I believe most of VideoOutlet etc. is already made by you in that PR.

Yes, I'm quite busy atm, but I should be able to finish the macOS texture stuff soon.

I've already refactored the video output handling so it's the responsibility of the platform-specific plugin to attach a "video output" to the actual player now. The Windows-specific implementation also handles resolution changes properly now. See https://github.com/jnschulze/dart_vlc/commits/refactor-video-output-handling-r

**Comment by alexmercerind** [comment]
> > @jnschulze I've requested you for the review. I didn't know iOS can link libVLC that easy. Seems I should work on Android too.
> > I assume you're busy with your schedule & life. I just wanted to ask if we could just use `"BGRA"` for macOS #122 without having to add additional features of defining the color space etc. It would just be a easy go, I believe most of VideoOutlet etc. is already made by you in that PR.
> 
> Yes, I'm quite busy atm, but I should be able to finish the macOS texture stuff soon.
> 
> I've already refactored the video output handling so it's the responsibility of the platform-specific plugin to attach a "video output" to the actual player now. The Windows-specific implementation also handles resolution changes properly now. See [jnschulze/dart_vlc@`refactor-video-output-handling-r` (commits)](https://github.com/jnschulze/dart_vlc/commits/refactor-video-output-handling-r)

Should I merge this PR before your that PR? I don't think it be causing conflicts to you (or will it)?

Thanks for the support.

File: ios/dart_vlc.podspec [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      'DEFINES_MODULE' => 'YES',
+      'LIBRARY_SEARCH_PATHS[sdk=iphoneos*]' => '$(inherited) $(PODS_TARGET_SRCROOT)/deps/device',
+      'OTHER_LDFLAGS[sdk=iphoneos*]' => '$(inherited) -ldart_vlc_core -Wl,-force_load,${PODS_TARGET_SRCROOT}/deps/device/libdart_vlc_core.a',
+      'LIBRARY_SEARCH_PATHS[sdk=iphonesimulator*]' => '$(inherited) $(PODS_TARGET_SRCROOT)/deps/simulator',
+      'OTHER_LDFLAGS[sdk=iphonesimulator*]' => '$(inherited) -ldart_vlc_core -Wl,-force_load,${PODS_TARGET_SRCROOT}/deps/simulator/libdart_vlc_core.a',
+      'CLANG_CXX_LANGUAGE_STANDARD' => 'c++17',
+      'CLANG_CXX_LIBRARY' => 'libc++',
+    }
+    s.swift_version = '5.0'
+  end
```
Comment (jnschulze) [inline ios/dart_vlc.podspec:54]: nit: missing newline ;)

**Review by jnschulze — COMMENTED** [review]
LGTM

**Comment by jnschulze** [comment]
> > > @jnschulze I've requested you for the review. I didn't know iOS can link libVLC that easy. Seems I should work on Android too.
> > > I assume you're busy with your schedule & life. I just wanted to ask if we could just use `"BGRA"` for macOS #122 without having to add additional features of defining the color space etc. It would just be a easy go, I believe most of VideoOutlet etc. is already made by you in that PR.
> > 
> > 
> > Yes, I'm quite busy atm, but I should be able to finish the macOS texture stuff soon.
> > I've already refactored the video output handling so it's the responsibility of the platform-specific plugin to attach a "video output" to the actual player now. The Windows-specific implementation also handles resolution changes properly now. See [jnschulze/dart_vlc@`refactor-video-output-handling-r` (commits)](https://github.com/jnschulze/dart_vlc/commits/refactor-video-output-handling-r)
> 
> Should I merge this PR before your that PR? I don't think it be causing conflicts to you (or will it)?
> 
> Thanks for the support.

Looks like they're completely independent, so just merge this one.

**Pull request merged**